### ISSUE TITLE
Future incompatibilities in uart_xilinx

### ISSUE THREAD

**Issue by puzzlewolf** [duskmoon314/uart-rs#14]
Hi, I'm getting future-incompatibility warnings for `uart-xilinx 0.1.0` with  cargo 1.75.0-nightly (794d0a825 2023-10-03).

Here's the full text, as printed by `cargo report future-incompatibilities --id 1 --package uart_xilinx@0.1.0`.

```
The package `uart_xilinx v0.1.0` currently triggers the following future incompatibility lints:
> warning: trailing semicolon in macro used in expression position
>   --> /home/brei_no/.cargo/registry/src/index.crates.io-6f17d22bba15001f/uart_xilinx-0.1.0/src/uart_16550/registers.rs:5:71
>    |
> 5  |         unsafe { &mut *(($expr) as *mut super::registers::Registers) };
>    |                                                                       ^
>    |
>   ::: /home/brei_no/.cargo/registry/src/index.crates.io-6f17d22bba15001f/uart_xilinx-0.1.0/src/uart_16550/uart.rs:94:18
>    |
> 94 |             reg: cast!(base_address),
>    |                  ------------------- in this macro invocation
>    |
>    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>    = note: for more information, see issue #79813 <https://github.com/rust-lang/rust/issues/79813>
>    = note: `#[allow(semicolon_in_expressions_from_macros)]` on by default
>    = note: this warning originates in the macro `cast` (in Nightly builds, run with -Z macro-backtrace for more info)
> 
> warning: trailing semicolon in macro used in expression position
>    --> /home/brei_no/.cargo/registry/src/index.crates.io-6f17d22bba15001f/uart_xilinx-0.1.0/src/uart_16550/registers.rs:5:71
>     |
> 5   |         unsafe { &mut *(($expr) as *mut super::registers::Registers) };
>     |                                                                       ^
>     |
>    ::: /home/brei_no/.cargo/registry/src/index.crates.io-6f17d22bba15001f/uart_xilinx-0.1.0/src/uart_16550/uart.rs:119:20
>     |
> 119 |         self.reg = cast!(base_address);
>     |                    ------------------- in this macro invocation
>     |
>     = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>     = note: for more information, see issue #79813 <https://github.com/rust-lang/rust/issues/79813>
>     = note: `#[allow(semicolon_in_expressions_from_macros)]` on by default
>     = note: this warning originates in the macro `cast` (in Nightly builds, run with -Z macro-backtrace for more info)
> 
> warning: trailing semicolon in macro used in expression position
>   --> /home/brei_no/.cargo/registry/src/index.crates.io-6f17d22bba15001f/uart_xilinx-0.1.0/src/uart_lite/registers.rs:5:71
>    |
> 5  |         unsafe { &mut *(($expr) as *mut super::registers::Registers) };
>    |                                                                       ^
>    |
>   ::: /home/brei_no/.cargo/registry/src/index.crates.io-6f17d22bba15001f/uart_xilinx-0.1.0/src/uart_lite/uart.rs:40:18
>    |
> 40 |             reg: cast!(base_address),
>    |                  ------------------- in this macro invocation
>    |
>    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>    = note: for more information, see issue #79813 <https://github.com/rust-lang/rust/issues/79813>
>    = note: `#[allow(semicolon_in_expressions_from_macros)]` on by default
>    = note: this warning originates in the macro `cast` (in Nightly builds, run with -Z macro-backtrace for more info)
> 
> warning: trailing semicolon in macro used in expression position
>   --> /home/brei_no/.cargo/registry/src/index.crates.io-6f17d22bba15001f/uart_xilinx-0.1.0/src/uart_lite/registers.rs:5:71
>    |
> 5  |         unsafe { &mut *(($expr) as *mut super::registers::Registers) };
>    |                                                                       ^
>    |
>   ::: /home/brei_no/.cargo/registry/src/index.crates.io-6f17d22bba15001f/uart_xilinx-0.1.0/src/uart_lite/uart.rs:46:20
>    |
> 46 |         self.reg = cast!(base_address);
>    |                    ------------------- in this macro invocation
>    |
>    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>    = note: for more information, see issue #79813 <https://github.com/rust-lang/rust/issues/79813>
>    = note: `#[allow(semicolon_in_expressions_from_macros)]` on by default
>    = note: this warning originates in the macro `cast` (in Nightly builds, run with -Z macro-backtrace for more info)
> 
> warning: trailing semicolon in macro used in expression position
>    --> /home/brei_no/.cargo/registry/src/index.crates.io-6f17d22bba15001f/uart_xilinx-0.1.0/src/uart_lite/registers.rs:5:71
>     |
> 5   |         unsafe { &mut *(($expr) as *mut super::registers::Registers) };
>     |                                                                       ^
>     |
>    ::: /home/brei_no/.cargo/registry/src/index.crates.io-6f17d22bba15001f/uart_xilinx-0.1.0/src/uart_lite/uart.rs:184:18
>     |
> 184 |             reg: cast!(base_address),
>     |                  ------------------- in this macro invocation
>     |
>     = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>     = note: for more information, see issue #79813 <https://github.com/rust-lang/rust/issues/79813>
>     = note: `#[allow(semicolon_in_expressions_from_macros)]` on by default
>     = note: this warning originates in the macro `cast` (in Nightly builds, run with -Z macro-backtrace for more info)
> 
> warning: trailing semicolon in macro used in expression position
>    --> /home/brei_no/.cargo/registry/src/index.crates.io-6f17d22bba15001f/uart_xilinx-0.1.0/src/uart_lite/registers.rs:5:71
>     |
> 5   |         unsafe { &mut *(($expr) as *mut super::registers::Registers) };
>     |                                                                       ^
>     |
>    ::: /home/brei_no/.cargo/registry/src/index.crates.io-6f17d22bba15001f/uart_xilinx-0.1.0/src/uart_lite/uart.rs:190:20
>     |
> 190 |         self.reg = cast!(base_address);
>     |                    ------------------- in this macro invocation
>     |
>     = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>     = note: for more information, see issue #79813 <https://github.com/rust-lang/rust/issues/79813>
>     = note: `#[allow(semicolon_in_expressions_from_macros)]` on by default
>     = note: this warning originates in the macro `cast` (in Nightly builds, run with -Z macro-backtrace for more info)
```

**Comment by puzzlewolf** [comment]
Ah, nvm, it's already fixed on `main`. Would be great you could make a release, though :)

**Comment by duskmoon314** [comment]
I last used it a long time ago and forgot to publish it. 😢 

I just published v0.2.0 and hope it works well.

**Comment by puzzlewolf** [comment]
Seems to work well, our tests are fine :) Thank you very much :rocket:
